front/lib/api/user.ts

```diff
@@ -2,61 +2,43 @@ import type { UserMetadataType, UserType } from "@dust-tt/types";
 
 import type { Authenticator } from "@app/lib/auth";
 import { User, UserMetadata } from "@app/lib/models/user";
+import { UserResource } from "@app/lib/resources/user_resource";
 
 import { MembershipResource } from "../resources/membership_resource";
 
-export function renderUserType(user: User): UserType {
-  return {
-    sId: user.sId,
-    id: user.id,
-    createdAt: user.createdAt.getTime(),
-    provider: user.provider,
-    username: user.username,
-    email: user.email,
-    firstName: user.firstName,
-    lastName: user.lastName,
-    fullName: user.firstName + (user.lastName ? ` ${user.lastName}` : ""),
-    image: user.imageUrl,
-  };
-}
-
 /**
  * This function checks that the user had at least one membership in the past for this workspace
  * otherwise returns null, preventing retrieving user information from their sId.
  */
 export async function getUserForWorkspace(
   auth: Authenticator,
   { userId }: { userId: string }
-): Promise<UserType | null> {
+): Promise<UserResource | null> {
   const owner = auth.workspace();
   if (!owner || !(auth.isAdmin() || auth.user()?.sId === userId)) {
     return null;
   }
 
-  const user = await User.findOne({
-    where: {
-      sId: userId,
-    },
-  });
+  const user = await UserResource.fetchByExternalId(userId);
 
   if (!user) {
     return null;
   }
 
   const membership =
     await MembershipResource.getLatestMembershipOfUserInWorkspace({
-      user: renderUserType(user),
+      user,
       workspace: owner,
     });
 
   if (!membership) {
     return null;
   }
 
-  return renderUserType(user);
+  return user;
 }
 
-export async function deleteUser(user: UserType): Promise<void> {
+export async function deleteUser(user: UserResource): Promise<void> {
```

Comment:
This should live in the `UserResource` now.

Reply:
Done